src/stores/orderBookStore.ts

```diff
@@ -14,24 +15,57 @@ class OrderBookStore extends BaseStore {
     super(store);
   }
 
-  createOrders = ({ask, bid, id, price, timestamp}: any) => ({
-    ask,
-    bid,
-    id,
-    price,
-    timestamp: timestamp.toLocaleTimeString()
-  });
-
-  fetchAll = async () => {
-    const resp = await this.api.fetchAll();
-    runInAction(() => {
-      this.orders = resp.map(this.createOrders);
-    });
+  @action
+  addOrder = (order: OrderBookModel) => (this.orders = [...this.orders, order]);
+
+  fetchAll = () => {
+    const getOrders = async () => {
```

Comment:
honestly, `fetchAll` implementation is a bit messy :) will propose some improvements below

Reply:
it can be improved with functional approach like so https://repl.it/repls/WordyWideeyedCondor

this is not a final implementation, just want to share an idea